content/blog/2022/cq-syntax.md

```diff
@@ -0,0 +1,365 @@
+  **Be careful with this** --
+  elements might collapse entirely
+  if they can't get either a width or height value
+  from their contents.
+
+There is no `block-size` option.
+I won't go into all the reasons here,
+but it wasn't possible for browsers to implement,
+and there are fewer use-cases for it anyway.
+
+Over time,
+Media Queries have expanded
+to cover much more than viewport sizing --
+we can query user preferences,
+display quality,
+color-depth,
+interaction capabilities,
+and more.
+We expect a similar expansion
+with Container Queries.
+_Size queries_ will ship this summer,
+but _style queries_ are already well-defined,
+and ready for browsers to implement.
+We're also looking into a range of possible
+_state queries_ --
+for example,
+selecting elements inside a `position: sticky` container
+based on the stuck or unstuck 'state'
+of that container.
+
+[^style]: Chrome has started to implement
+  a prototype of style queries
+  (_for custom properties only_)
+  in Chrome Canary,
+  using the `--enable-blink-test-features`
+  runtime (command line) flag.
+  Keep an eye out for more details soon!
+
+## Naming containers with `container-name`
+
+By default,
+a container query will look for
+_the nearest ancestor container of the appropriate type_.
+So if we ask about the `inline-size` of a container,
+we'll get the inline dimensions of
+the nearest ancestor with a size-based `container-type`.
+That's a handy default,
+but it's more resilient if we can name our containers,
+and know that we're measuring the right thing.
+
+We can do that with the `container-name` property.
+It allows us to give a container
+_any number of custom names_ that we want.
+Those names can be any valid CSS 'custom-ident'
+that we come up with
+(similar to naming keyframe animations, grid-areas, layers, etc),
+and they are not required to be unique.
+There are only a few reserved words --
+like `none` or `not` or `initial` --
+that we can't use as names.
+
+Think of container-names more like a 'class' than an 'ID'.
+We can make them as broad or specific as we need.
+I think it's useful to establish
+some broad categories like
+`layout` or `component` --
+and then occasionally add more
+detailed/specific identifiers as needed
+(like `main` in the code sample above).
+
+## The `container` shorthand
+
+There's also a `container` shorthand
+that allows us to define both the container
+types and names in a single property.
+This should only be used
+to set both the name **AND** type of a property:
+
+```css
+main, .sidebar {
+  /* <name> comes before the slash, <type> comes after */
+  container: page-layout / inline-size;
+}
+```
+
+Note that _names go first_, before the slash.
+**This is one of the big changes
+that will break a lot of demos.**
+In an earlier draft of the feature,
+the order was reversed.
+If you see an old container query
+article or talk or demo
+that no longer works,
+_this is probably why_.
```

Comment:
Do we have a callout for this kind of stuff? It feels like this is what people will want to skim to if they are looking for the change

Reply:
We don't yet. We could maybe use something similar to the callouts that David added to css-sandbox? See https://css.oddbird.net/sass/color-spaces/proposal/ for example - with both a 'note' and a 'warning' version of the callout. @SondraE maybe you want to weigh in on that?

Reply:
@mirisuzanne @stacyk Great idea to add a design element to highlight that info! Makes sense to me to use what David used. Or maybe with slight adjustments in this direction. https://zenplates.co/patterns/callouts